stripe-ui-core/src/main/java/com/stripe/android/uicore/elements/CheckboxFieldController.kt

```diff
@@ -0,0 +1,60 @@
+package com.stripe.android.uicore.elements
+
+import androidx.annotation.RestrictTo
+import androidx.annotation.StringRes
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusDirection
+import com.stripe.android.uicore.R
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class CheckboxFieldController constructor(
+    val labelResource: LabelResource? = null,
+    val debugTag: String = "",
+    initialValue: Boolean = false,
+) : SectionFieldErrorController, SectionFieldComposable {
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    class LabelResource(
+        @StringRes val labelId: Int,
+        vararg args: Any
+    ) {
+        val formatArgs = args
+    }
+
+    private var hasBeenEdited = false
+    private val _isChecked = MutableStateFlow(initialValue)
+    val isChecked: StateFlow<Boolean>
+        get() = _isChecked
+
+    override val error: Flow<FieldError?> = _isChecked.map { value ->
```

Comment:
Do we need the error? If the user unchecks a required checkbox, we would just disable the primary button, right?

Reply:
This was my initial thinking, but web flow also produces an error state for a checkbox element as well. Maybe we can just make it an optional parameter of the controller
<img src="https://github.com/stripe/stripe-android/assets/141707240/665227b2-ac93-43f3-b952-8c077eed6521" width="40%" height="40%" />

Reply:
If we want to reuse this in the future, then we should likely make showing an error an optional parameter. I’m fine leaving it as-is for the BACS use case.